client/src/utils/functions.js

```diff
@@ -14,3 +14,38 @@ export function toIso(europeanDate) {
   const [day, month, year] = europeanDate.split("/");
   return `${year}-${month}-${day}`;
 }
+
+export function budgetFormValidator(formData) {
+  const errors = {};
+  const specialCharacters = /[^A-Za-zÀ-ÿ0-9 ]/;
+
+  if (!formData.name.length) {
+    errors.NameError =
+      "Le nom ne peut pas être vide ou composé uniquement d'espaces. Les espaces en début et fin de nom ne comptent pas comme un caractère.";
+  }
+  if (formData.name) {
+    if (formData.name.length > 100) {
+      errors.NameError = "Nom trop long : maximum 100 caractères.";
+    }
+    if (formData.name.length < 3) {
+      errors.NameError =
+        "Nom trop court : 3 caractères minimum. Les espaces en début et fin de nom ne comptent pas comme un caractère.";
```

Comment:
**thought**: I feel like it could be wrapped in a same condition with a message like "Le nom doit être compris entre 3 et 100 caractères" instead of two conditions, what do you think ?

Reply:
I thought about that too, but was wondering if max character is relevant for the user to know. By that i mean, except one user that want to limit test my input, who would create a 100 characters budget's name ?
But yeah, i could put the two conditions together.

Reply:
Okay, I do see the logic behind this, you can keep it like that if you want :)